lib/route/route.js

```diff
@@ -472,25 +532,122 @@ const checkPageIdParams = (alpha, omega) => {
   return true
 }
 
-const getPreviousPage = (currentPage, userData) => {
+/*
+ *  This is a weird function. I'm not sure how to refactor it (yet) so I'm commenting it up. Hello
+ *
+ *  It does one or both of two things:
+ *
+ *  First, it asks whether or not the current page is a step of an upload journey. If it is: the previous
+ *  page is the parent of the step
+ *
+ *  Second, and if it is not a step of an upload journey, it rewinds and fast-forwards again up to
+ *  the previous page
+ *
+ *  I've tried to do some renaming of variables to make their content a bit clearer. Hope it helps
+ */
+function getPreviousPage (currentPage, userData) {
   const { _id } = currentPage
-  let startPage
-  let instance = getInstance(_id)
-  while (instance._parent) {
-    startPage = instance._parent
-    instance = getInstance(instance._parent)
-  }
 
-  let candidatePage = { _id: startPage }
-  while (candidatePage) {
-    const candidateResultPage = getNextPage(candidatePage, userData, true)
+  /*
+   *  No _id, no previous page
+   */
+  if (_id) {
+    let currentInstance = getInstance(_id)
+
+    /*
+     *  Determine whether the current page is a check or summary step
+     */
+    if (isUploadCheckPage(currentInstance) || isUploadSummaryPage(currentInstance)) {
```

Comment:
The logic here and at 612 is exactly the same so it's a-function we go (on Monday/post sleep)